substratevm/src/com.oracle.svm.driver/src/com/oracle/svm/driver/BundleSupport.java

```diff
@@ -182,6 +313,192 @@ static BundleSupport create(NativeImage nativeImage, String bundleArg, NativeIma
+            } else if(containerTool.equals("docker") && !isRootlessDocker()) {
+                throw NativeImage.showError("Only rootless docker is supported for containerized builds.");
+            }
+            containerToolVersion = getContainerToolVersion(containerTool);
+
+            if(bundleContainerTool != null) {
+                if (!containerTool.equals(bundleContainerTool)) {
+                    NativeImage.showWarning(String.format("The given bundle file %s was created with container tool '%s' (using '%s').", bundleFileName, bundleContainerTool, containerTool));
+                } else if (containerToolVersion != null && bundleContainerToolVersion != null && !containerToolVersion.equals(bundleContainerToolVersion)) {
+                    NativeImage.showWarning(String.format("The given bundle file %s was created with different %s version '%s' (installed '%s').", bundleFileName, containerTool, bundleContainerToolVersion, containerToolVersion));
+                }
+            }
+        } else {
+            for(String tool : SUPPORTED_CONTAINER_TOOLS) {
+                if(isToolAvailable(tool)) {
+                    if(tool.equals("docker") && !isRootlessDocker()) {
+                        nativeImage.showMessage(BUNDLE_INFO_MESSAGE_PREFIX + "Rootless context missing for docker.");
+                        continue;
+                    }
+                    containerTool = tool;
+                    containerToolVersion = getContainerToolVersion(tool);
+                    break;
+                }
+            }
+            if (containerTool == null) {
+                throw NativeImage.showError(String.format("Please install one of the following tools before running containerized native image builds: %s", SUPPORTED_CONTAINER_TOOLS));
+            }
+        }
+
+        int exitStatusCode = createContainer();
+        switch (ExitStatus.of(exitStatusCode)) {
+            case OK -> fetchContainerEnvironment();
+            case BUILDER_ERROR ->
+                /* Exit, builder has handled error reporting. */
+                    throw NativeImage.showError(null, null, exitStatusCode);
+            case OUT_OF_MEMORY -> {
+                nativeImage.showOutOfMemoryWarning();
+                throw NativeImage.showError(null, null, exitStatusCode);
+            }
+            default -> {
+                String message = String.format("Container build request for '%s' failed with exit status %d",
+                        nativeImage.imageName, exitStatusCode);
+                throw NativeImage.showError(message, null, exitStatusCode);
+            }
+        }
+    }
+
+    private void fetchContainerEnvironment() {
+        ProcessBuilder pb = new ProcessBuilder(containerTool, "run", "--rm", containerImage, "bash", "-c", "/usr/bin/env");
+        Process p = null;
+        try {
+            p = pb.start();
+            p.waitFor();
+            (new BufferedReader(new InputStreamReader(p.getInputStream())))
```

Comment:
Use try-with-resources
```
try (var in = new BufferedReader(new InputStreamReader(p.getInputStream()))) {
    in.lines()
    ...
}
```

Reply:
Addressed in commit https://github.com/oracle/graal/pull/6496/commits/d3afef2a54b0c2cb0de7a1aeab35546da3477a03